mllib/src/test/scala/org/apache/spark/ml/classification/LogisticRegressionSuite.scala

```diff
@@ -2567,10 +2504,13 @@ class LogisticRegressionSuite
     val model1 = lr.fit(smallBinaryDataset)
     val lr2 = new LogisticRegression().setInitialModel(model1).setMaxIter(5).setFamily("binomial")
     val model2 = lr2.fit(smallBinaryDataset)
-    val predictions1 = model1.transform(smallBinaryDataset).select("prediction").collect()
-    val predictions2 = model2.transform(smallBinaryDataset).select("prediction").collect()
-    predictions1.zip(predictions2).foreach { case (Row(p1: Double), Row(p2: Double)) =>
-      assert(p1 === p2)
+    val binaryExpected = model1.transform(smallBinaryDataset).select("prediction").collect()
+      .map(_.getDouble(0))
+    for (model <- Seq(model1, model2)) {
```

Comment:
This line code `val binaryExpected = model1.transform(smallBinaryDataset).select("prediction").collect().map(_.getDouble(0))` used to generate the `binaryExpected` dataset.
And then test model1/model2 on both df.transform and streamDF.transform and compare result to `binaryExpected` (assert equal).
Otherwise we need to hardcoding the `binaryExpected` dataset in the code.

Reply:
My thought is that testing binaryExpected (from model1) against model2 would already test the 2 things we care about:
* batch vs streaming prediction
* initial model

I'll just merge this though since it's not a big deal (just a bit longer testing time).